src/NUnitFramework/framework/Constraints/EqualConstraintResult.cs

```diff
@@ -73,7 +73,7 @@ private void DisplayDifferences(MessageWriter writer, object? expected, object?
                 DisplayEnumerableDifferences(writer, expectedEnumerable, actualEnumerable, depth);
             else if (expected is Stream expectedStream && actual is Stream actualStream)
                 DisplayStreamDifferences(writer, expectedStream, actualStream, depth);
-            else if (_tolerance is not null)
+            else if (_tolerance != Tolerance.Default)
```

Comment:
There is no longer a need for this test and you can simply call:
`writer.DisplayDifferences(expected, actual, _tolerance);` unconditionally.